pallets/pools/src/tests.rs

```diff
@@ -803,24 +803,21 @@ fn submission_period() {
 			]
 		));
 
-		// Can't submit the same solution twice
-		assert_err!(
-			Pools::submit_solution(
-				pool_owner_origin.clone(),
-				0,
-				vec![
-					TrancheSolution {
-						invest_fulfillment: Perquintill::one(),
-						redeem_fulfillment: Perquintill::from_float(0.01),
-					},
-					TrancheSolution {
-						invest_fulfillment: Perquintill::one(),
-						redeem_fulfillment: Perquintill::one(),
-					}
-				]
-			),
-			Error::<Test>::NotNewBestSubmission
-		);
+		// Can submit the same solution twice
```

Comment:
Maybe you could add a test case specifically for the deadlock scenario? Basically copying https://github.com/centrifuge/tinlake/blob/develop/src/lender/test/coordinator-submitEpoch.sol#L250 and/or https://github.com/centrifuge/tinlake/blob/develop/src/lender/test/coordinator-submitEpoch.sol#L290

Reply:
Not sure I am getting, that this is the deadlock scenario. With the current `>=` there can not be a deadlock as assured by the test above. But I can add a test that allows to submit a zero-solution as the first one, to trigger the `ChallangePeriod`. WDYT?

Reply:
Ah, yes. That makes total sense. I will append this scenario to this pr.